Flow.Rmd

```diff
@@ -27,26 +36,63 @@ flow_theme <- theme_bw() + theme(
+For more references on flow cytometry analysis and applications, see:
 
+- [Practical Flow Cytometry](https://www.wiley.com/en-us/Practical+Flow+Cytometry%2C+4th+Edition-p-9780471411253) [@Shapiro1994-ri]: An in-depth text with details on advanced topics such as cell sorting, single molecule detection, and applications of flow in hematology, immunology, microbiology, food science, bioterrorism, and more
+- [Translational Applications of Flow Cytometry in Clinical Practice](http://www.jimmunol.org/content/188/10/4715) [@Jaye2012-vc]: An overview of clinical applications in cancer and immunological disease diagnosis as well as cell therapy
+- [Critical assessment of automated flow cytometry data analysis techniques](https://www.nature.com/articles/nmeth.2365) [@Aghaeepour2013-xo]: A presentation of results from an analytical competition ([FlowCAP](http://flowcap.flowsite.org/)) held to assess the performance of algorithmic advancements in flow cytometry data processing
 
 ## Exploring FlowRepository
 
-TBD
+While many sources of public flow cytometry data exist, the subject of this chapter will be experiments uploaded to [FlowRepository](https://flowrepository.org/).  Specifically, the [FlowRepositoryR](https://bioconductor.org/packages/release/bioc/html/FlowRepositoryR.html) package will be used to download two datasets related to ["Optimized Multicolor Immunophenotyping Panels" (OMIP)](https://fluorofinder.com/omips/) publications.  These panels consist of dyes and antibodies used to label samples relevant for one specific experimental purpose such as characterizing natural killer cell populations or chemokine expression on helper T cells.  
+
+Data available from these publications may be found in FlowRepository simply by searching for studies containing the keywork "OMIP".  Using the FlowRepository site is generally the simplest way to find relevant studies but programmatic access can also be helpful for doing more targeted searches such as for studies that have an associated FlowJo workspace, as shown below.
+
+```{r flow-repository-search-example, results='asis'}
+# Search for all studies matching the OMIP keyword (which returns a list of string IDs) and then
+# fetch all corresponding details for the first 25
+studies <- flowRep.search('OMIP') %>% sort %>% head(25) %>% map(flowRep.get)
+
+# Study objects are implemented as S4 classes with id and name slots as well as an attachments slot, 
+# which is a list further containing S4 objects (with a "name" slot having values like "20151210_Workspace.jo")
+tibble(
+  id=studies %>% map_chr(~ .@id),
+  name=studies %>% map_chr(~ .@name),
+  panel=name %>% str_extract('(?<=OMIP-)\\d+') %>% as.integer,
+  has_flowjo_ws=studies %>% map_lgl(~ map(.@attachments, ~.@name) %>% str_detect('\\.jo|\\.wsp') %>% any)
+) %>% 
+  filter(has_flowjo_ws) %>% 
+  select(-has_flowjo_ws) %>% 
+  arrange(panel) %>%
+  knitr::kable()
```

Comment:
I think this table (and others) need a [horizontal scroll box](https://cran.r-project.org/web/packages/kableExtra/vignettes/awesome_table_in_html.html#scroll_box) to see all columns.

Reply:
Ah very cool, did not realize you could do that with knitr.  I'll make sure to add it.